src/data-workspace/form-wrapper.js

```diff
@@ -21,29 +20,34 @@ function createInitialValues(dataValueSet) {
     ])
 }
 
-const onSubmit = (values, form) => console.log({ values, form })
-
-const subscriptions = {}
-
-export function FinalFormWrapper({ children, dataValueSet }) {
+export function FormWrapper({ children, dataValueSet, validFormKey }) {
     // This prevents rerendering on every data value update, but also prevents
     // reinitializing (and populating) an empty form when going back online.
     // TODO: Reinitialize form `onSuccess` of dataValueSets query
     // See https://dhis2.atlassian.net/browse/TECH-1357
     const [initialValues] = useState(() => createInitialValues(dataValueSet))
-    return (
-        <Form
-            onSubmit={onSubmit}
-            initialValues={initialValues}
-            subscriptions={subscriptions}
-            keepDirtyOnReinitialize
-            mutators={{ setFieldData }}
-        >
-            {() => children}
-        </Form>
+
+    const setInitialDataValues = useValueStore(
+        (state) => state.setInitialDataValues
     )
+
+    const initialValuesInStore =
+        useValueStore((state) => state.getInitialDataValues())?.formKey ===
+        validFormKey
+
+    useLayoutEffect(() => {
```

Comment:
I don't think it matters too much here, but I think normal `useEffect` should suffice.